### PR TITLE
First part of the occultist totem/monolith update. DO NOT MERGE!!

### PR DESCRIPTION
**Pull request by SigDoesCode**
## About The Pull Request
This update adds new structures for the Occultist to create: totems and monoliths.

First up, totems. Totems are smaller structures that come from a T1 power, and they can be generated as many times as you have points to spend. They (are supposed to, as of 10-5-21 I'm not sure this is working right) drain sanity about 2.5 times as fast as a holoplant restores it. To offset this, destroying them restores 10 sanity to anyone who can see it. 

Totems are randomized. They have random names, random descriptions, and random icons. Their descriptions are in two parts, a lore dump and instructions. The lore is a general flavor description and the instruction is what tool you need to take it apart. If you use the wrong tool, it doesn't work. The list of tool qualities required is as follows:
Bolt turning, cutting, drilling, hammering, prying, pulsing, sawing, screw driving, welding, wire cutting.

This is an example totem. 
image
The first sentence in the description is a flavor text. The second sentence hints at what tool you need to use to deconstruct it.
These are the totem icons.
image

Up next are monoliths. Monoliths are massive structures that are intended to be the center to an occultist's base. Once every twenty minutes, the occultist can touch the monolith to cause all non-occultists who can see it to instantly break down (though as of 10-5-21 this is causing two breakdowns at once but I'm not sure this is an issue, it's only once every 20 minutes.) 

Monoliths also have a random name, icon, and description, but there are also some lore-important ones. Those have a set name and description. On spawning it, it'll first pick if it's generating a lore-important one and then continue generating from there. They cannot be taken apart (as of 10-5-21, still taking suggestions for it, though I'm thinking a randomly generated list of long steps for deconstruction, so you can't just slap it down in combat or something.) 

This is what it looks like when someone touches it. ANYONE can trigger the breakdowns, not just an occultist.
image

Additionally, trying to touch the monolith before it's recharged will give you a timer.
image

The icons for monoliths:
image
Currently Podium is unused because I can't figure out a satisfactory sprite. 

And and example monolith:
image
This one is lore important! 
image
This one is not.

## Why It's Good For The Game
Occultists get fun stuff! Also hopefully this makes them feel weirder and gives the crew more opportunity to play with "discovering" occultists without knowing exactly who it is, as they're created without fingerprints.

## Changelog
```changelog
add: Added totems and monoliths for occultists to spawn.
spellcheck: Changed an occultist power description to match what it actually does now. 
```

### CODE REVIEW

**Comment by SigDoesCode** [comment]
Please, please, PLEASE do not merge this yet. I'm putting this up for a bit of review and because it's almost done.

**Comment by SigDoesCode** [comment]
Second push in. Things that are changed from the first push:
1. Deconstructing totems no longer tries to spawn materials. I can't get this to work for the life of me.
2. Monoliths are no longer guaranteed to give a negative breakdown. I can't get them to actually affect the person they're spawned in. 

Things I want feedback on specifically:
1. Are the descriptions for the deconstruction tools for the totems alright? Are they too on the nose? Too cryptic?
2. Are the icons okay? 

If everything is good I'll fix up the PR properly and make the push.

**Comment by SigDoesCode** [comment]
Thank you for the labels, Shadz, you're awesome.

**Comment by SigDoesCode** [comment]
Okay, they should be ready for a proper review now. Things changed from the original PR:
- The totem power now costs 0 madness to pick
- Totems down from 5 san damage to 2.5
- No longer sets your environmental sanity cap to 0, whoops.

**Comment by Jamini** [comment]
This branch is out of date, failing CI is expected here for run linters.

**Comment by Shadow-Quill** [comment]
> This branch is out of date, failing CI is expected here for run linters.

Linters will only fail on code that has been touched, because of how git works. (As you can see with the lint.)
Review comments imminent.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_powers.dm [hunk 1]
```diff
@@ -8,6 +8,57 @@
...
+	if(stat == UNCONSCIOUS)
+		to_chat(src, "You cannot construct things while unconsious.")
+		return
+	if(spendpoints(0)) //Free, but needs stuff.
+		var/list/stuff = src.loc.contents
+		var/list/roaches = list() //Needs one roach of MOB_MEDIUM or higher or five roaches of MOB_SMALL or lower.
+		var/large_roach = FALSE //For storing if we have a large roach or not
+		var/obj/item/weapon/flame/candle/candle = null //Checking if we have a candle
+		for(var/subject in stuff) //Check for roach
+			//to_world(R.type) //DEBUG??? I don't know how this is broken, R.type isn't returning anything help
```
Comment (Shadow-Quill) [inline zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_powers.dm:37]: Debug

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_powers.dm [hunk 2]
```diff
@@ -8,6 +8,57 @@
...
+			if(istype (subject, /mob/living/carbon/superior_animal/roach))
```
Comment (Shadow-Quill) [inline zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_powers.dm:38]: Style!~ (bit picky, I'm aware.)
```suggestion
			if(istype(subject, /mob/living/carbon/superior_animal/roach))
```

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_powers.dm [hunk 3]
```diff
@@ -611,4 +683,15 @@
...
 	//Hood
 	to_chat(usr, SPAN_DANGER("Your eyes have melted, now you can see. Tear open reality and release what is underneath."))
-	usr.replace_in_slot(new /obj/item/clothing/head/space/occulthood, slot_head, skip_covering_check = TRUE)
\ No newline at end of file
+	usr.replace_in_slot(new /obj/item/clothing/head/space/occulthood, slot_head, skip_covering_check = TRUE)
+
+/datum/power/occultist/monolith
+	name = "Unnatural Sculpting"
+	desc = "Craft a monolith to your madness and show others the truth."
+	activecost= 0
```
Comment (Shadow-Quill) [inline zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_powers.dm:691]:
```suggestion
	activecost = 0
```

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]
Overall, looks okay (code-wise) to me. Do have a few comments though.

File: zzzz_modular_occulus/code/datums/perks/oddity.dm [hunk 4]
```diff
@@ -10,4 +10,9 @@
...
 	name = "Empath"
 	desc = "You have a sixth sense about people. You can tell when they are stressed."
-	icon_state = "flowers"
\ No newline at end of file
+	icon_state = "flowers"
+
+/datum/perk/oddity/surefooting
+	name = "placeholder"
+	desc = "placeholder"
+	icon_state = "null"
```
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/datums/perks/oddity.dm]: Couldn't remove this comment after seeing the update.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+	if(deconqual == QUALITY_PULSING)
+		descdecon = "It's held together by some electronic mechanism."
+	if(deconqual == QUALITY_SAWING)
+		descdecon = "The frame looks thin. A saw might help."
+	if(deconqual == QUALITY_SCREW_DRIVING)
+		descdecon = "There are very obvious screws holding the whole thing together."
+	if(deconqual == QUALITY_WELDING)
+		descdecon = "It's poorly welded together."
+	if(deconqual == QUALITY_WIRE_CUTTING)
+		descdecon = "Wire holds the structure together."
```
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:76]: I don't know what an associated list is. You'd need to explain it like I'm a five year old and it'd probably break the code again. Honestly, after three months I don't care if it's perfect or not. You can change it to an associated list if you want, but I don't want to break it after promising it for this long.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+	density = TRUE
+	use_power = NO_POWER_USE
+	unacidable = 1
+	anchored = TRUE
+	var/maker
+	var/builder //for holding the builder of the totem for seeing sanity later?
+	var/sanrestore = 10 //How much sanity is restored to non-occultists who watch this thing getting taken apart?
+	var/desclore //For holding the lore portion of the description
+	var/decontime //For holding the time that the totem takes to be deconstructed
+	var/descdecon //Holding the decon portion of the description
```
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:16]: It's there for futureproofing and my brain wanting to maybe add more things that play with those vars. Why do you want it defined under new?

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+		if(name == "\proper a vessel-shaped hole")
+			desc = "A pool-shaped hole where your life will arrive and disappear. Despite the fact there is nothing here, it doesn't matter; \
+			 the objects matter much less than the holes they occupy. This would be comforting, if only you understood."
+		if(name == "\proper Avimelech")
+			desc = "It was so beautiful, once. It believed so strongly in the idealism that rested in all living things; it was your greatest ally. \
+			And this is how you repay a lost soul trying to get home?"
+		if(name == "\proper THE SKIES")
+			desc = "THE SKIES ARE BURIED DEEP, AND WHAT IS ABOVE AWAITS BELOW, IF ONLY YOU COULD DRILL THROUGH HELL."
+		if(name == "\proper OBJECT/NAME")
+			desc = "As the world closes and opens again, the only thing you will carry is the knowledge of death and dreams."
```
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:216]:
```
if(prob(80)) //generic
	name = pick("yadda", "dabba", "doo")
	desc = pick("Meet the flintstones.", "Not your average bear", "What's up doc?")

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+
+//Set the look. Put these in zzzz_modular_occulus/icons/turf/occultist/misc.dmi
+	icon_state = pick(\
+		"totem",
+		"totem2",
+		"totem3",
+		"totem4",
+		"totem5",
+		"totem6"\
+		)
```
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:102]: Does that work?

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+				P.forceMove(loc)
+				component_parts -= P
+				*/
+			qdel(src)
+			return
+	else to_chat(user, SPAN_NOTICE("Your tool bends away from the totem unnaturally."))
+
+//The processing object to handle sanity lowering below the environmental cap.
+/obj/machinery/occultist/totem/Process()
+	for(var/mob/living/carbon/human/H in view(src.loc))
```
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:145]: Yes.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 7 above)
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:216]: See my other comment on associated lists.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+
+//Can't make this find the owner of the sanity thing where the breakdown spawns. It's giving a "can't find owner" runtime on click. Undefined var in breakdowns.dm, 407, 307, 47, 53
+//The above is still current as of 10/14/21, why does nothing make sense here -Sigma
+//Todo: add sound effects!
+
+/obj/machinery/occultist/monolith/attack_hand(mob/user as mob)
+	if(charge > 0) //Can we activate?
+		charge-- //Subtract the charge
+		for(var/mob/living/carbon/human/viewer in view(src.loc)) //Track all things looking at it.
+			if(viewer.sanity) //If they have a brain
```
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:234]: Why?

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm [hunk 11]
```diff
@@ -0,0 +1,278 @@
...
+/obj/machinery/occultist/monolith //A massive grandiose centerpiece to an occultist's lair.
+	name = "improper monolith"
+	desc = "A basic monolith to something that never existed. Ahelp if you see this."
+	icon = 'zzzz_modular_occulus/icons/turf/occult/totems.dmi'
+	icon_state = "kubrick"
+	density = TRUE	//no walking through this
+	use_power = NO_POWER_USE //not a machine proper, like the nuke
+	unacidable = 1 //yeah we don't want it getting eaten, only one spawnable per occultist per round
+	anchored = TRUE
+	var/cooldown = 20 MINUTES //For the touch verb. 20 minutes in ticks is 20,000
```
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:161]: I want it saved for future use and other additions. This way, I can easily find and set the cooldown.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 8 above)
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:102]: Mhm, other objects do something similar for slight icon changes, such as
`airlock[open : "" ? "_open"]` or similar. (Though, that particular example may not exist.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm [hunk 12]
```diff
@@ -0,0 +1,278 @@
...
+		time = ((recharge - world.time)/10) //Determine how long is actually left on the recharge and turn it into a big number of seconds.
+		min = (FLOOR((time/60), 1)) //Turn the time var into minutes. We don't round this because it's just for display, the seconds come later.
+		//sec = (round((min % time), 0.1)) //This should be telling you how many seconds are left until the minute ticks over, but seconds are stupid, figure this out later.
+
+		to_chat(user, SPAN_NOTICE("Even an insane mind realizes the danger in delving too deep too quickly. Wait [min] minutes for your mind to steel itself once more."))
+			//Todo: add "and [sec] seconds" after "minutes" in the description once it's checking seconds correctly. Currently it's just outputting minutes again. -Sigma 10-4-21
+
+	else if(charge < 0) //debug in case this breaks.
+		to_chat(user, SPAN_NOTICE("This shouldn't happen, you've got fewer than 0 charges in the monolith, ahelp."))
+
```
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:275]: Done!

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 10 above)
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:234]: Lowers how far other things are indented, which generally helps with overall readability.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:16]: Well, as of right now, they aren't used anywhere else except setting the description, so there's no need to have them set on the obj itself

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 5 above)
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:76]: Not going to be a sticking point (and I'll resolve the other comment about this), but something for someone else to potentially keep in mind.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 8 above)
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:102]: This works, fixed.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:16]: Except that they can be used for future ideas that I've had and I don't want to have to rewrite it again.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 11 above)
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:154]: Alright.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+		"Your bones hum in harmony with the rumbling of the ship. Let them out.",
+		"It moves when it thinks you aren't looking at it.",
+		"There is a hole in your head where your mind should be, but you don't feel any different.",
+		"Is belief the key? It- no, it's too late for it to matter."\
+		)
+	//Now we slap the description together. First the lore then how to take it apart.
+	desc = "[desclore] [descdecon]"
+
+//Set the look. Put these in zzzz_modular_occulus/icons/turf/occultist/misc.dmi
+	icon_state = pick("totem[rand(1,6)]")
```
Comment (Shadow-Quill) [inline zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:95]: Do note, that the `totem` icon_state will have to be changed to `totem1`, if it wasn't already.

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 10 above)
[the remark this answers is not in the document]
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm]: This is breaking something when I try. If you want to fix it for readability, go ahead, but it's working and it needed to be ready like two months ago.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 13 above)
Comment (SigDoesCode) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:95]: Fixed.

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm:16]: Alright

File: zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm (hunk 10 above)
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/game/gamemodes/occultist/occultist_machines.dm]: Alright

**Review by Shadow-Quill — APPROVED** [review]
Code looks fine to me. I do welcome others checking over stuff as well.

**Pull request merged**